### DISCUSSION TITLE
Support for FaceID / TouchID / Passcode to improve Security 🔐

### DISCUSSION THREAD

**Discussion by michaeldlfx** [community/community#3450] [Mobile] [answered]
Hey 👋 I love using the app, but one thing that I'm missing is the possibility to enable FaceID / TouchID / Passcode.

I'd personally toggle this on to prompt me with one of the aforementioned options whenever the app is (re-)opened. 

Especially since the app has read & write access 😅 (which I like, reviewing & merging PR's on the go is brilliant).

Thanks!

**Comment by t12344** [comment]
> Hey 👋 I love using the app, but one thing that I'm missing is the possibility to enable FaceID / TouchID / Passcode.
> 
> I'd personally toggle this on to prompt me with one of the aforementioned options whenever the app is (re-)opened. 
> 
> Especially since the app has read & write access 😅 (which I like, reviewing & merging PR's on the go is brilliant).
> 
> Thanks!

**Reply by michaeldlfx** [reply]
@candyho any chance this is being worked on? 🙂

**Reply by inikolaev** [reply]
This would be a really helpful feature. I use the app on an iPad sometimes, which is generally used by my daughter, but wouldn't like her to accidentally open it and approve a PR on my behalf :)

**Reply by azuraCoomsatay** [reply]
Bingindi

**Comment by t12344** [comment]
> > Hey 👋 I love using the app, but one thing that I'm missing is the possibility to enable FaceID / TouchID / Passcode.
> > 
> > I'd personally toggle this on to prompt me with one of the aforementioned options whenever the app is (re-)opened. 
> > 
> > Especially since the app has read & write access 😅 (which I like, reviewing & merging PR's on the go is brilliant).
> > 
> > Thanks!
> 
>

**Comment by williamhorning** [comment]
Touch ID/ Face ID can be used as a security key for 2FA on Github, but prompting you for it on opening the app doesn’t exist but could be added.

**Reply by hfhbd** [reply]
Yes please. And Safari too https://webkit.org/blog/11312/meet-face-id-and-touch-id-for-the-web/

**Comment by aconsuegra** [comment, marked as answer]
Thanks for the feedback and the feature request!

We are evaluating it this feature and will plan to add it to the mobile apps in the future.

**Reply by michaeldlfx** [reply]
Great thanks! Any update on this? 🙂

**Reply by FynnZW** [reply]
Is this still planned? Would be a great feature

**Reply by Richiedevop** [reply]
![IMG_20240213_021548_273.jpg](https://github.com/community/community/assets/156444805/2fe82177-59cb-4bfa-8b03-6d6447c5e06f)


```

**Reply by shahinkhorshidfar** [reply]
[]()

**Reply by Dolordolordolor** [reply]
😅

**Comment by neatcoder** [comment]
After 2 years, no progress on this important security feature?

**Comment by m-tse** [comment]
+1

I consider GitHub even more sensitive than my credit cards/bank accounts, because code is intellectual property, which is invaluable. I had to delete the GitHub app because of the lack of this feature. Would love to reinstall if it was added.

Namely, I'm afraid of the "thief steals unlocked phone out of my hands" type attack.

**Comment by dennemark** [comment]
I have just opened a Github Mobile App went to a repository and started an action. The action led to a deploy of a whole website. I could easily do this, because the Mobile App does not have any security layer when opening it.
Fortunately, the app and phone were my own, so I knew what I was doing.
Please...add this feature!

**Comment by imanmahjoubi** [comment]
@dennemark @m-tse @neatcoder @FynnZW @michaeldlfx @williamhorning @t12344  Thank you for the feedback everyone ! App lock via biometric is coming very soon and we would like to keep getting your feedback to improve the feature.

**Comment by aconsuegra** [comment]
This feature is now available on GitHub Mobile! Check the announcement in https://github.blog/changelog/2024-08-06-app-lock-via-face-id-or-biometrics/, and let us know if you have any feedback. Thanks for your patience!
